backend/routes/users.js

```diff
@@ -1,9 +1,252 @@
+		logger.error("Username and password are required");
+		return res
+			.status(400)
+			.json({ message: "Username and password are required" });
+	}
+
+	const Users = await dataSource.getRepository(User);
+
+	try {
+		if (!(await Users.findOne({ where: { id: 1 } }))) {
+			// If user with id 1 does not exist, register a new user
+			await registerUser();
+		}
+		// Check if any user exists in the database with the provided username
+		const existingUser = await Users.findOne({ where: { userName } });
+		if (existingUser) {
+			logger.error("Unauthorized");
+			// If user exists, compare the provided password with the hashed password from the database
+			const match = await bcrypt.compare(password, existingUser.password);
+			if (!match) {
+				return res.status(401).json({ message: "Unauthorized" });
+			}
+			const token = jwt.sign(
+				{ id: existingUser.id, userName: existingUser.userName },
+				JWT_SECRET,
+				{},
+			);
+			res.cookie("token", token);
+			logger.info("User successfully logged in");
+			return res
+				.status(200)
+				.json({ message: "User successfully logged in", user: existingUser });
+		} else {
+			logger.error("Unauthorized");
+			return res.status(401).json({ message: "Unauthorized" });
+		}
+	} catch (error) {
+		logger.error("Internal server error:", error);
+		res.status(500).json({ message: "Internal server error" });
+	}
+});
+
+/**
+ * POST /api/users/logout
+ *
+ * Purpose: Log out a user by clearing the JWT token cookie.
+ *
+ * Pre-conditions: None
+ *
+ * Post-conditions:
+ * - Clears the token cookie.
+ * - Returns status code 204 on successful logout.
+ * - If an internal server error occurs, returns a message "Internal server error" with status code 500.
+ */
+// POST route for user logout
+router.post("/logout", async (req, res) => {
+	try {
+		// Clear the token cookie
+		res.clearCookie("token");
+		// Log successful logout
+		logger.info("User successfully logged out");
+		// Send a success response
+		res.sendStatus(204);
+	} catch (error) {
+		// Log any errors that occur during logout
+		logger.error("Error during logout:", error);
+		// Send an error response
+		res.status(500).json({ message: "Internal server error" });
+	}
+});
+
+/**
+ * GET /api/users
+ *
+ * Purpose: Retrieve user details using a JWT token.
+ *
+ * Pre-conditions:
+ * - Request must contain a valid JWT token in the cookie named 'token'.
+ *
+ * Post-conditions:
+ * - If a valid token is provided, returns user details with status code 200.
+ * - If no token is provided, returns null with status code 200.
+ * - If an error occurs during token verification, returns a message "Internal server error" with status code 500.
+ */
+// GET route to fetch the single user's details
+router.get("/", async (req, res) => {
+	const { token } = req.cookies;
+	if (token) {
+		jwt.verify(token, JWT_SECRET, {}, (err, user) => {
+			if (err) {
+				logger.error("Error verifying JWT:", err);
+				throw err;
```

Comment:
This throws an error (which effectively crashes the whole backend) if the user does not have a token. Please handle this more gracefully.

Reply:
I don't know if this is where the exact error originated from, but a more detailed outlook of the issue is issue #92

Reply:
Due to the need for this branch to allow developers to start development for ID4, we will leave this issue for a future patch. (issue #92)